sycl/tools/sycl-ls/sycl-ls.cpp

```diff
@@ -363,8 +366,22 @@ int main(int argc, char **argv) {
     if (DiscardFilters && FilterEnvVars.size())
       unsetFilterEnvVars();
 
+    // Store the original UR_LOG_LOADER environment variable and enable printing
+    // of any errors related to adapter loading.
+    const char *orig_ur_log_loader_var = std::getenv("UR_LOG_LOADER");
+    std::string orig_ur_log_loader_var_str;
+    if (orig_ur_log_loader_var != NULL)
+      orig_ur_log_loader_var_str.assign(orig_ur_log_loader_var);
```

Comment:
Why do you need `std::string` at all?

Reply:
I need to have a deep copy of UR_LOG_LOADER value, not just a pointer